examples/resources/infisical_cert_manager_certificate_profile/resource.tf

```diff
@@ -0,0 +1,193 @@
+  country      = "US"
+  locality     = "San Francisco"
+  province     = "California"
+}
+
+resource "infisical_cert_manager_internal_ca_intermediate" "issuing" {
+  project_slug = infisical_project.pki.slug
+  parent_ca_id = infisical_cert_manager_internal_ca_root.root.id
+
+  name          = "enterprise-issuing-ca"
+  common_name   = "Enterprise Issuing Certificate Authority"
+  organization  = "Example Corp"
+  ou            = "IT Security"
+  country       = "US"
+  locality      = "San Francisco"
+  province      = "California"
+  key_algorithm = "RSA_2048"
+}
+
+resource "infisical_cert_manager_certificate_template" "web_server" {
+  project_slug = infisical_project.pki.slug
+
+  name        = "Web Server Template"
+  description = "Template for web server certificates"
+
+  subject {
+    type     = "common_name"
+    allowed  = ["*.example.com", "*.internal.example.com"]
+    required = ["*.example.com"]
+  }
+
+  subject {
+    type     = "organization"
+    required = ["Example Corp"]
+  }
+
+  subject {
+    type     = "country"
+    required = ["US"]
+  }
+
+  sans {
+    type     = "dns_name"
+    allowed  = ["*.example.com", "*.internal.example.com"]
+    required = ["*.example.com"]
+  }
+
+  sans {
+    type    = "email"
+    allowed = ["*@example.com"]
+  }
+
+  key_usages {
+    allowed = ["digital_signature", "key_encipherment"]
+  }
+
+  extended_key_usages {
+    allowed = ["server_auth", "client_auth"]
+  }
+
+  validity {
+    max = "90d"
+  }
+
+  algorithms {
+    signature     = ["SHA256-RSA", "SHA256-ECDSA", "SHA384-ECDSA"]
+    key_algorithm = ["RSA-2048", "RSA-3072", "ECDSA-P256", "ECDSA-P384"]
+  }
+}
+
+# API enrollment profile for web server certificates
+resource "infisical_cert_manager_certificate_profile" "web_server_api" {
+  project_slug            = infisical_project.pki.slug
+  ca_id                   = infisical_cert_manager_internal_ca_intermediate.issuing.id
+  certificate_template_id = infisical_cert_manager_certificate_template.web_server.id
+
+  name            = "web-server-api"
+  description     = "API enrollment for web server certificates"
+  enrollment_type = "api"
+  issuer_type     = "ca"
+
+  api_config {
+    auto_renew        = true
+    renew_before_days = 7
+  }
+}
+
+# EST enrollment profile for web server certificates
+resource "infisical_cert_manager_certificate_profile" "web_server_est" {
+  project_slug            = infisical_project.pki.slug
+  ca_id                   = infisical_cert_manager_internal_ca_intermediate.issuing.id
+  certificate_template_id = infisical_cert_manager_certificate_template.web_server.id
+
+  name            = "web-server-est"
+  description     = "EST enrollment for web server certificates"
+  enrollment_type = "est"
+  issuer_type     = "ca"
+
+  est_config {
+    passphrase                      = var.est_passphrase
+    disable_bootstrap_ca_validation = false
+    ca_chain                        = var.encrypted_ca_chain
+  }
+}
+
```

Comment:
IGNORE PLACEMENT OF COMMENT

I think we should add some sort of validation so that users don't accidentally configure the key usages and extended key usages field like the ff:

<img width="625" height="471" alt="image" src="https://github.com/user-attachments/assets/a78b422e-9d9b-4b0a-a8c2-82424354a513" />

The problem is that issuing a certificate from the profile becomes problematic because of the conflict:
<img width="1512" height="855" alt="image" src="https://github.com/user-attachments/assets/9bcc4a83-7408-4b6d-ad91-2ceb078fd664" />